test/open_mic_test.rb

```diff
@@ -0,0 +1,44 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/joke'
+require './lib/user'
+require './lib/open_mic'
+
+class OpenMicTest < Minitest::Test
+  def test_it_exists
+    open_mic = OpenMic.new({location: "Comedy Works", date: "11-20-18"})
+    assert_instance_of OpenMic, open_mic
+  end
+
+  def test_it_has_attributes
+    open_mic = OpenMic.new({location: "Comedy Works", date: "11-20-18"})
+    assert_equal open_mic.location, "Comedy Works"
+    assert_equal open_mic.date, "11-20-18"
+  end
+
+  def test_it_can_welcome_user
+    open_mic = OpenMic.new({location: "Comedy Works", date: "11-20-18"})
+    sal = User.new("Sal")
+    ali = User.new("Ali")
+    open_mic.welcome(sal)
+    open_mic.welcome(ali)
+    assert_equal 2, open_mic.performers.length
+  end
+
+  def test_for_repeated_jokes
+    open_mic = OpenMic.new({location: "Comedy Works", date: "11-20-18"})
+    sal = User.new("Sal")
+    ali = User.new("Ali")
+    open_mic.welcome(sal)
+    open_mic.welcome(ali)
+    ali.learn(joke_1)
+    ali.learn(joke_2)
+    refute open_mic.repeated_jokes?
+    sal.learn(joke_1)
+    assert open_mic.repeated_jokes?
```

Comment:
nice job writing this test first, even though you didn't quite get to the implementation of the method.